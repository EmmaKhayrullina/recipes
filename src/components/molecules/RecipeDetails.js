import React from 'react';
import styled from 'styled-components';
import Title from '../atoms/Title';

const Container = styled.div`
  padding: 15px;
  overflow: hidden;
  background-color: var(--second-color);
`;

const Details = styled.div`
  min-width: calc(100% - 240px);
  float: left;
`;

const SubTitle = styled.h2`
  margin: 0 0 10px;
`;

const Text = styled.p`
  margin: 0 0 20px;
  white-space: pre-line;
`;

const Photo = styled.figure`
  display: flex;
  width: 200px;
  margin: 20px 0 20px 20px;

  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
`;

const RecipeDetails = ({ recipe }) => {
  return (
    <article data-testid="test-recipe-details">
      <Title>{recipe.title}</Title>

      <Container>
        <Details>
          <SubTitle>Ingredients:</SubTitle>
          <Text>{recipe.ingredients}</Text>

          <SubTitle>Description:</SubTitle>
          <Text>{recipe.description}</Text>
        </Details>

        {recipe.image && (
          <Photo>
            <img src={recipe.image.url} alt="recipe" />
          </Photo>
        )}
      </Container>
    </article>
  );
};

export default RecipeDetails;
