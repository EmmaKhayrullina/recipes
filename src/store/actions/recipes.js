import { UPDATE_RECIPE, ADD_RECIPE, REMOVE_RECIPE, FETCH_RECIPES } from './actionTypes';

export const fetchRecipes = recipes => {
  return {
    type: FETCH_RECIPES,
    payload: recipes,
  };
};

export const addRecipe = recipe => {
  return {
    type: ADD_RECIPE,
    payload: recipe,
  };
};

export const updateRecipeFields = fields => {
  return {
    type: UPDATE_RECIPE,
    payload: fields,
  };
};

export const removeRecipe = id => {
  return {
    type: REMOVE_RECIPE,
    payload: id,
  };
};
