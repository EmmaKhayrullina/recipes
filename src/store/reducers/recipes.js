import { FETCH_RECIPES, ADD_RECIPE, REMOVE_RECIPE, UPDATE_RECIPE, RESET_APP } from '../actions/actionTypes';

const recipesReducer = (state = [], action) => {
  switch (action.type) {
    case FETCH_RECIPES:
      return action.payload;
    case ADD_RECIPE:
      return [...state, action.payload];
    case UPDATE_RECIPE:
      return state.map(recipe => {
        return recipe.id === action.payload.id ? { ...recipe, ...action.payload } : recipe;
      });
    case REMOVE_RECIPE:
      return state.filter(recipe => recipe.id !== action.payload);
    case RESET_APP:
      return [];
    default:
      return state;
  }
};

export default recipesReducer;
