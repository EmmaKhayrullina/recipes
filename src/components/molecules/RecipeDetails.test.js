import React from 'react';
import { render, screen } from '@testing-library/react';
import RecipeDetails from './RecipeDetails';
import recipes from '../../__mocks__/recipesData';

describe('RecipeDetails component', () => {
  const recipe = recipes[1];

  test('render recipe details', () => {
    // Act
    render(<RecipeDetails recipe={recipe} />);

    // Assert
    expect(screen.getByTestId('test-recipe-details')).toMatchSnapshot();
  });
});
