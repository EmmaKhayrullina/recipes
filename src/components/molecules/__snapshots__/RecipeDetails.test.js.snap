// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetails component render recipe details 1`] = `
<article
  data-testid="test-recipe-details"
>
  <h1
    class="sc-AxjAm kqDOSG"
  >
    Pumpkin soup
  </h1>
  <div
    class="sc-AxirZ FOtfm"
  >
    <div
      class="sc-AxiKw hkJvxZ"
    >
      <h2
        class="sc-AxhCb kXrYeE"
      >
        Ingredients:
      </h2>
      <p
        class="sc-AxhUy igyvKH"
      >
        pumpkin potato onion carrot
      </p>
      <h2
        class="sc-AxhCb kXrYeE"
      >
        Description:
      </h2>
      <p
        class="sc-AxhUy igyvKH"
      >
        Cook
      </p>
    </div>
    <figure
      class="sc-AxgMl fQtgwy"
    >
      <img
        alt="recipe"
        src="https://recipeImage"
      />
    </figure>
  </div>
</article>
`;
